cmake/GenerateVersionFile.cmake

```
@@ -2,3 +2,26 @@ configure_file(
   ${CMAKE_SOURCE_DIR}/cmake/version.cpp.in
   ${CMAKE_BINARY_DIR}/src/ccache/version.cpp
   @ONLY)
+
+if(WIN32)
+
+  if(CCACHE_VERSION MATCHES "^([0-9]+)([0-9]+)([0-9]+)$")
```

Should this be something like

```suggestion
  if(CCACHE_VERSION MATCHES "^([0-9]+)\.([0-9]+)\.([0-9]+)\$")
```

?